iac/terraform/aws/donor_search_sf/stepFunction.tf

```diff
@@ -24,43 +24,57 @@ resource "aws_cloudwatch_log_group" "donor_search_state_machine_logs" {
   retention_in_days = 90
 }
 
-resource "aws_iam_policy" "step_function_logging_policy" {
-  name        = "${var.environment}-donor-search-sf-logging-policy"
-  description = "Policy to allow Step Functions to write logs to CloudWatch"
-
-  policy = <<EOF
-{
-  "Version": "2012-10-17",
-  "Statement": [
-    {
-      "Effect": "Allow",
-      "Action": [
-        "logs:CreateLogDelivery",
-        "logs:GetLogDelivery",
-        "logs:UpdateLogDelivery",
-        "logs:DeleteLogDelivery",
-        "logs:ListLogDeliveries",
-        "logs:PutResourcePolicy",
-        "logs:DescribeResourcePolicies",
-        "logs:DescribeLogGroups"
-      ],
-      "Resource": "*"
-    }
-  ]
+resource "aws_iam_role_policy_attachment" "step_function_attach_policy" {
+  policy_arn = aws_iam_policy.step_function_policy.arn
+  role       = aws_iam_role.step_function_role.name
 }
-EOF
+
+resource "aws_iam_policy" "step_function_policy" {
+  name        = "${var.environment}-donor-search-sf-policy"
+  description = "Policy to allow Step Functions to write logs to CloudWatch"
+  policy      = data.aws_iam_policy_document.step_function_policy_document.json
 }
 
-resource "aws_iam_role_policy_attachment" "attach_logging_policy" {
-  policy_arn = aws_iam_policy.step_function_logging_policy.arn
-  role       = aws_iam_role.step_function_role.name
+data "aws_iam_policy_document" "step_function_policy_document" {
+  statement {
+    actions = [
+      "dynamodb:Query",
+      "dynamodb:GetItem"
+    ]
+    effect    = "Allow"
+    resources = [var.dynamodb_table_arn]
+  }
+
+  statement {
+    actions = [
+      "logs:CreateLogDelivery",
+      "logs:GetLogDelivery",
+      "logs:UpdateLogDelivery",
+      "logs:DeleteLogDelivery",
+      "logs:ListLogDeliveries",
+      "logs:PutResourcePolicy",
+      "logs:DescribeResourcePolicies",
+      "logs:DescribeLogGroups"
+    ]
+    effect    = "Allow"
+    resources = ["*"]
```

Comment:
Why is this a *?

Reply:
this is suggested for step function https://docs.aws.amazon.com/step-functions/latest/dg/cw-logs.html#cloudwatch-iam-policy